### PR TITLE
ng_pktbuf: handle size == pkt->size case for ng_pktbuf_add()

### PR DESCRIPTION
**Pull request by authmillenon**
Fixes an issue mentioned in https://github.com/RIOT-OS/RIOT/pull/3367#issuecomment-120917530.

### CODE REVIEW

File: sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c [hunk 1]
```diff
@@ -258,6 +258,13 @@ static ng_pktsnip_t *_pktbuf_add_unsafe(ng_pktsnip_t *pkt, void *data,
             return NULL;
         }
 
+        if (size == pkt->size) {
+            DEBUG("pktbuf: size (%u) == pkt->size (%u) => just set new packet "
+                  "type and return\n", (unsigned)size, (unsigned)pkt->size);
+            pkt->type = type;
+            return pkt;
```
Comment (daniel-k) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: You forgot to free `new_pktsnip` if I'm not mistaken.

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: Ah yes.

Comment (authmillenon) [inline sys/net/crosslayer/ng_pktbuf/ng_pktbuf.c]: Fixed

**Comment by daniel-k** [comment]
Now `new_pktsnip` is unitialized in [this](https://github.com/RIOT-OS/RIOT/pull/3375/files#diff-bf20997ebd03be1bed686194b825ad63R232) block.

**Comment by authmillenon** [comment]
That comes from doing fixes during meetings :D

**Comment by daniel-k** [comment]
Looks good and seems to work, so ACK :)

**Comment by authmillenon** [comment]
Squashed.

**Comment by authmillenon** [comment]
Travis is happy (strider isn't, but that's not important now).

**Pull request merged**